python-package/xgboost/spark/core.py

```diff
@@ -315,6 +320,23 @@ def _validate_params(self):
                     )
 
 
+def _validate_and_convert_feature_col_as_float_col_list(
+    dataset, features_col_names: list
+) -> list:
+    """Values in feature columns must be integral types or float/double types"""
+    feature_cols = []
+    for c in features_col_names:
+        if isinstance(dataset.schema[c].dataType, DoubleType):
+            feature_cols.append(col(c).cast(FloatType()).alias(c))
```

Comment:
I don't think we need to do the cast here.  XGBoost can handle double type.

Reply:
casting here can reduce the shuffle write size.

Reply:
Q:
Do we need to support integer type here ?